### ISSUE TITLE
Error when building

### ISSUE THREAD

**Issue by 123yamsha** [cisco/ChezScheme#275]
When building threaded version (unthreaded builds and installs normally) I encouter this error (needless to say that libz.a is present):

```
(cd ta6le && make install)
(cd c ; make)
gcc  -m64 -msse2 -Wpointer-arith -Wall -Wextra -Werror -Wno-implicit-fallthrough -O2 -D_REENTRANT -pthread  -c -DX86_64 -I../boot/ta6le -I../zlib scheme.c
(cd ../zlib; make)
Please use ./configure first.  Thank you.
ld -melf_x86_64 -r -X -o ../boot/ta6le/kernel.o statics.o segment.o alloc.o symbol.o intern.o gcwrapper.o gc-ocd.o gc-oce.o number.o schsig.o io.o new-io.o print.o fasl.o stats.o foreign.o prim.o prim5.o flushcache.o schlib.o thread.o expeditor.o scheme.o i3le.o ../zlib/libz.a
ld: cannot find ../zlib/libz.a: No such file or directory
make[2]: *** [Makefile:34: ../boot/ta6le/kernel.o] Error 1
make[1]: *** [Makefile:20: build] Error 2
make: *** [Makefile:22: install] Error 2

```

**Comment by akeep** [comment]
Hmm... I tried to reproduce this, but I wasn't able to.  I've occasionally seen something like this if I had built a different version first, though I've had trouble reproducing it, even when I've had it happen to me.

**Comment by istolpov** [comment]
I have the same issue, building on WSL2 Ubuntu 22.04.2 LTS. Tried to build `v9.5.8`.
I tried running `./configure` and even `make` manually in zlib, but the error still persists.

After some `make clean`-s and `git clean --force`-s it just worked. The only thing I've changed was add '--thread' option in the end. Not sure if that helps or not but probably needs to be investigated further.

Also, make sure you do the `make install` as a superuser (`sudo`).
